next-react-query-tailwind/src/pages/auth/signin.tsx

```diff
@@ -0,0 +1,50 @@
+/* eslint-disable @next/next/no-img-element */
+import { CtxOrReq, getProviders, signIn } from 'next-auth/client';
+import { InferGetServerSidePropsType } from 'next';
+
+export default function SignIn({
+  providers,
+}: InferGetServerSidePropsType<typeof getServerSideProps>) {
+  return (
+    <>
+      <section className="flex flex-col items-center w-full h-full bg-gray-900">
+        {providers
+          ? Object.values(providers).map((provider) => (
+              <div
+                key={provider.name}
+                className="flex justify-center content-center h-5/6 w-11/12"
+              >
+                <button
+                  className="self-center w-3/5 lg:w-1/5 py-5 px-2 text-white border-solid border-white border-2   rounded-sm hover:text-gray-900 hover:bg-white transition-colors"
+                  onClick={() => signIn(provider.id, { callbackUrl: '/' })}
+                >
+                  Sign in with {provider.name}
+                </button>
+              </div>
+            ))
+          : ''}
```

Comment:
just a nit pick I would use null instead of an empty string but I'm not sure it makes a difference in practice